packages/frontend/app/lib/apollo.server.ts

```diff
@@ -0,0 +1,35 @@
+import { ApolloClient, InMemoryCache } from '@apollo/client'
+import type { NormalizedCacheObject } from '@apollo/client'
+
+/* eslint-disable no-var */
+declare global {
+  var __apolloClient: ApolloClient<NormalizedCacheObject> | undefined
```

Comment:
Looked into this. I know with some libraries there is an issue where multiple instances get spawned due to HMR and cache clearing. There is no information/bug related to this type of issue when ApolloClient is used inside Remix. I'll remove it from the global namespace.

Reply:
Remix doesn't have HMR yet, so the dev server will restart entirely when you make changes. This is bad for db clients etc, because it'll make a new connection to the db. They solve this in [their example](https://remix.run/docs/en/1.14.1/tutorials/jokes#md-connect-to-the-database) tutorial by making the client global. I would probably leave the client in the global namespace.